IdentityCore/src/telemetry/request_telemetry/MSIDCurrentRequestTelemetry.m

```diff
@@ -29,17 +29,70 @@ @implementation MSIDCurrentRequestTelemetry
 
 - (NSString *)telemetryString
 {
-    return @"";
+    return [self serializeCurrentTelemetryString];
 }
 
-- (instancetype)initWithTelemetryString:(__unused NSString *)csvString error:(__unused NSError **)error
+- (instancetype)initWithTelemetryString:(NSString *)telemetryString error:(NSError **)error
 {
     self = [super init];
     if (self)
     {
+        NSError *internalError;
+        [self deserializeCurrentTelemetryString:telemetryString error:&internalError];
         
+        if (internalError) {
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"Failed to initialize server telemetry, error: %@", MSID_PII_LOG_MASKABLE(internalError));
+            if (error) *error = internalError;
+            return nil;
+        }
     }
     return self;
 }
 
+#pragma mark - Private
+
+-(NSString *)serializeCurrentTelemetryString
+{
+    int forceRefreshValue = (self.forceRefresh ? 1 : 0);
+    NSString *telemetryString = [NSString stringWithFormat:@"%ld|%ld,%d|", self.schemaVersion, self.apiId, forceRefreshValue];
+    
+    // Make sure string to be returned is less than 4kB in size
+    if ([telemetryString lengthOfBytesUsingEncoding:NSUTF8StringEncoding] > 4000)
+    {
+        return nil;
+    }
+    
+    return telemetryString;
+}
+
+-(void)deserializeCurrentTelemetryString:(NSString *)telemetryString error:(NSError **)error
```

Comment:
Just curious, why do we need to deserialize telemetry?

Reply:
My understanding is that a telemetry string can be saved to the device, so there needs to be logic for initializing a telemetry object from a string in that situation. This would mainly happen for the Last Request telemetry string in the case where an error occurred previously, so the saved telemetry string from that error request would be used to initialize the object. I believe initialization from a telemetry string was added to the protocol for this scenario, so I added an implementation for Current Request as well, although I'm not sure about when it would be used.

@antrix1989 may be able to provide more detail.

Reply:
Is there a way to achieve it without having to parse strings :)?